backend/pkg/api/data_access/vdb_rewards.go

```diff
@@ -71,63 +74,41 @@ func (d *DataAccessService) GetValidatorDashboardRewards(ctx context.Context, da
-		BlocksScheduled       uint64          `db:"blocks_scheduled"`
-		BlocksProposed        uint64          `db:"blocks_proposed"`
-		SyncScheduled         uint64          `db:"sync_scheduled"`
-		SyncExecuted          uint64          `db:"sync_executed"`
-		SlashedInEpoch        uint64          `db:"slashed_in_epoch"`
-		SlashedAmount         uint64          `db:"slashed_amount"`
-	}{}
-
-	queryParams := []interface{}{}
-	rewardsQuery := ""
-
 	groupIdSearchMap := make(map[uint64]bool, 0)
 
-	rewardsDataQuery := `
-		SUM(COALESCE(e.attestations_reward, 0) + COALESCE(e.blocks_cl_reward, 0) +
-		COALESCE(e.sync_rewards, 0) + COALESCE(e.slasher_reward, 0)) AS cl_rewards,
-		COALESCE(SUM(COALESCE(r.value, ep.fee_recipient_reward * 1e18)), 0) AS el_rewards,		
-		SUM(COALESCE(e.attestations_scheduled, 0)) AS attestations_scheduled,
-		SUM(COALESCE(e.attestations_executed, 0)) AS attestations_executed,
-		SUM(COALESCE(e.blocks_scheduled, 0)) AS blocks_scheduled,
-		SUM(COALESCE(e.blocks_proposed, 0)) AS blocks_proposed,
-		SUM(COALESCE(e.sync_scheduled, 0)) AS sync_scheduled,
-		SUM(COALESCE(e.sync_executed, 0)) AS sync_executed,
-		SUM(CASE WHEN e.slashed_by IS NOT NULL THEN 1 ELSE 0 END) AS slashed_in_epoch,
-		SUM(COALESCE(s.slashed_amount, 0)) AS slashed_amount
-		`
+	// ------------------------------------------------------------------------------------------------------------------
+	// Build the query that serves as base for both the main and EL rewards queries
+	ds := goqu.Dialect("postgres").
+		Select(
+			goqu.L("e.epoch")).
+		From(goqu.L("validator_dashboard_data_epoch e")).
+		Where(goqu.L("e.epoch > ?", latestFinalizedEpoch-epochLookBack))
 
 	if dashboardId.Validators == nil {
-		queryParams = append(queryParams, dashboardId.Id, latestFinalizedEpoch-epochLookBack)
-		whereQuery := fmt.Sprintf("WHERE v.dashboard_id = $%d AND e.epoch > $%d", len(queryParams)-1, len(queryParams))
+		ds = ds.
+			InnerJoin(goqu.L("users_val_dashboards_validators v"), goqu.On(goqu.L("e.validator_index = v.validator_index"))).
+			Where(goqu.L("v.dashboard_id = ?", dashboardId.Id))
+
 		if currentCursor.IsValid() {
+			condition := ""
 			if currentCursor.IsReverse() {
 				if currentCursor.GroupId == t.AllGroups {
 					// The cursor is on the total rewards so get the data for all groups excluding the cursor epoch
-					queryParams = append(queryParams, currentCursor.Epoch)
-					whereQuery += fmt.Sprintf(" AND e.epoch%[1]s$%[2]d", sortSearchDirection, len(queryParams))
+					condition = fmt.Sprintf("e.epoch %s ?", sortSearchDirection)
 				} else {
 					// The cursor is on a specific group, get the data for the whole epoch since we could need it for the total rewards
-					queryParams = append(queryParams, currentCursor.Epoch)
-					whereQuery += fmt.Sprintf(" AND e.epoch%[1]s=$%[2]d", sortSearchDirection, len(queryParams))
+					condition = fmt.Sprintf("e.epoch %s= ?", sortSearchDirection)
 				}
+				ds = ds.Where(goqu.L(condition, currentCursor.Epoch))
```

Comment:
maybe set the `ds.Where` directly in the if branches, skipping the condition variable?